### PR TITLE
test: Add negative test cases to TargetsManagementResourceIT

### LINKED ISSUE #910: test: Add negative test cases to TargetsManagementResourceIT
## Description

The TargetsManagementResourceIT integration tests added in PR #908 currently only cover happy path scenarios. To improve test coverage and ensure proper error handling, we need to add negative test cases.

## Expected Test Cases

- Invalid/malformed payload handling
- Validation error responses (missing required fields, invalid values)
- Edge cases (null values, empty strings, invalid port ranges)
- Invalid service type handling
- Unauthorized access attempts
- Non-existent target operations

## Additional Context

Reference: PR #908 - "test(integration): Add Targets Management API integration tests"

This is a good first issue for contributors looking to improve test coverage.

### PR DESCRIPTION
**Pull request by DeepsanBhandari**
Closes #910
Adds negative (error-path) integration tests to `TargetsManagementResourceIT` 
as requested in issue #910. The existing tests from PR #908 only covered happy 
path scenarios. This PR improves test coverage by validating that the API 
correctly rejects invalid, malformed, and unauthorized requests.

### 1. Malformed / Invalid Payload
- `testAddTargetMalformedJson` — non-JSON body should return 400
- `testAddTargetEmptyBody` — empty `{}` body should return 400

### 2. Missing Required Fields
- `testAddTargetMissingHost` — missing host field should return 400
- `testAddTargetMissingPort` — missing port field should return 400
- `testAddTargetMissingService` — missing service field should return 400

### 3. Invalid Values / Edge Cases
- `testAddTargetPortTooHigh` — port 99999 (above valid range) should return 400
- `testAddTargetNegativePort` — port -1 should return 400
- `testAddTargetEmptyHost` — empty string host should return 400
- `testAddTargetNullHost` — null host should return 400

### 4. Invalid Service Type
- `testAddTargetInvalidServiceType` — unknown service name should return 400
- `testAddTargetEmptyServiceType` — empty service string should return 400

### 5. Non-Existent Target Operations
- `testRemoveNonExistentTarget` — removing unknown target should return 404
- `testListTargetsForUnknownService` — listing unknown service should return 404

### 6. Unauthorized Access
- `testAddTargetUnauthorized` — add without auth header should return 401
- `testRemoveTargetUnauthorized` — remove without auth header should return 401

## Notes

- Tests extend `WanakuIntegrationBase` and return an empty list from 
  `activeWanakuDownstreamServices()` since no downstream services are needed
- Unauthorized tests (14, 15) are only meaningful when Keycloak auth is 
  enabled in the test profile
- Some scenarios (e.g. port range validation) may require server-side 
  validation to be added before these tests fully pass

## Summary by Sourcery

Add negative integration coverage for the targets management REST API to verify correct handling of invalid and unauthorized requests.

Tests:
- Introduce TargetsManagementResourceIT integration test class covering malformed payloads, missing fields, invalid values, and unknown services for target creation and management endpoints.
- Add integration tests asserting 4xx responses for operations on non-existent targets and for requests made without authentication.

### CODE REVIEW

File: tests/integration/src/test/java/ai/wanaku/mcp/TargetsManagementResourceIT.java [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+/**
+ * Negative (error-path) integration tests for the Targets Management REST API.
+ *
+ * <p>These tests verify that the API correctly rejects invalid, malformed, or
+ * unauthorized requests and returns appropriate HTTP error status codes.
+ *
+ * <p>Reference: Issue #910. Related PR: #908.
+ */
+@QuarkusTest
+public class TargetsManagementResourceIT extends WanakuIntegrationBase {
```
Comment (orpiske) [inline tests/integration/src/test/java/ai/wanaku/mcp/TargetsManagementResourceIT.java:34]: This file should extend `WanakuRouterTest` instead of `WanakuIntegrationBase`

File: tests/integration/src/test/java/ai/wanaku/mcp/TargetsManagementResourceIT.java [hunk 2]
```diff
@@ -0,0 +1,299 @@
+/*
```
Comment (orpiske) [inline tests/integration/src/test/java/ai/wanaku/mcp/TargetsManagementResourceIT.java:1]: The file is in wrong module. It should be a part of the wanaku-router-backend module.

File: tests/integration/src/test/java/ai/wanaku/mcp/TargetsManagementResourceIT.java [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+
+/**
+ * Negative (error-path) integration tests for the Targets Management REST API.
+ *
+ * <p>These tests verify that the API correctly rejects invalid, malformed, or
+ * unauthorized requests and returns appropriate HTTP error status codes.
+ *
+ * <p>Reference: Issue #910. Related PR: #908.
+ */
+@QuarkusTest
```
Comment (orpiske) [inline tests/integration/src/test/java/ai/wanaku/mcp/TargetsManagementResourceIT.java:33]: Look at the annotations in DiscoveryResourceTest and use the same.

**Comment by orpiske** [comment]
Is this still active?

**Comment by orpiske** [comment]
This one seems to have stalled. Let's close it without merge (we can reopen if it's not the case - no harm in doing that).
